### DISCUSSION TITLE
Winget repo integration?

### DISCUSSION THREAD

**Discussion by FalsePhilosopher** [chocolatey/choco#2882] [Ideas] [answered]
why is this feature request needed?
To diversify the choco package ecosystem with an alt repo to pull from. It being MS stamped it's more corporate IT friendly sounding then using a 3rd party repo such as choco. No more switching between choco and winget

What is the benefit of being able to install a Winget manifest as a Chocolatey package?
Not switch between searching winget first then searching choco for a package, when choco could search both and give you the option to install from winget's repo or choco's repo or using MS's GUI for winget and just use choco's gui for everything. It wouldn't be just a package but more of a plugin as it would need to be part of choco's search/update functionality.

If you can provide specific examples, that would really help.
How Discover the package manager for the linux DE KDE works, here is a screen shot of how the settings look like
https://github.com/FalsePhilosopher/Test/blob/main/screenshot.jpg
You install the flatpak, and snap discover frontend packages and discover now one click updates all my other repos and backends of the different packaging systems out there(flatpak's and snaps are like exe vs msi and traditional packages are like bats so you can see why we use flatpak and snap over source built packages much like msi and exe's over bat's).

NA

NA

**Comment by gep13** [comment]
@FalsePhilosopher thank you for joining the Chocolatey Community here on GitHub!

In order to best help answer the question, we need to make sure that we fully understand the request that you are making.  Can you please go back and edit this issue using the `Enhancement / Feature Request` template from the list of templates that you would have been presented with. This will hopefully allow us to fully understand exactly what it is you are requested, because as it stands, I am not exactly clear.

For example, why is this feature request needed? What is the benefit of being able to install a Winget manifest as a Chocolatey package?  If you can provide specific examples, that would really help. We would need to know the answer to these types of questions before we could comment further.

**Comment by FalsePhilosopher** [comment]
@gep13 Sorry i'm new to github, and I read the doc's and could not figure out how to get the Enhancement / Feature Request template and only the bug report template. But to answer your questions

why is this feature request needed?
To diversify the choco package ecosystem with an alt repo to pull from. It being MS stamped it's more corporate IT friendly sounding then using a 3rd party repo such as choco.

What is the benefit of being able to install a Winget manifest as a Chocolatey package?
Not switch between searching winget first then searching choco for a package, when choco could search both and give you the option to install from winget's repo or choco's repo or using MS's GUI for winget and just use choco's gui for everything. It wouldn't be just a package but more of a plugin as it would need to be part of choco's search/update functionality.

If you can provide specific examples, that would really help.
How Discover the package manager for the linux DE KDE works, here is a screen shot of how the settings look like
https://github.com/FalsePhilosopher/Test/blob/main/screenshot.jpg
You install the flatpak, and snap discover frontend packages and discover now one click updates all my other repos and backends of the different packaging systems out there(flatpak's and snaps are like exe vs msi and traditional packages are like bats so you can see why we use flatpak and snap over source built packages much like msi and exe's over bat's).

**Comment by gep13** [comment]
@FalsePhilosopher Thank you for the additional information.  I have edited your original post to include the template, if you could go back and edit it with your additional information, and anything else that you can think of, that would be great.

**Comment by gep13** [comment, marked as answer]
> @FalsePhilosopher said...
To diversify the choco package ecosystem with an alt repo to pull from.

The slight problem here is that while Chocolatey is a package manager, and uses a package repository of packages to install from, Winget isn't strictly speaking a package manager, it installs application from a GitHub repository, using manifest files.  There are two fundamentally different things.

That is not to say that it isn't possible to integrate Winget into Chocolatey, we already have some examples of what we call alternative sources, where you can install, for example, Windows Features using the source parameter `--source windows features`.  I could see Winget being added in a similar way, but there would be lots of things that would need to be discussed/stubbed out here before we could move forward with it.

**Comment by FalsePhilosopher** [comment]
@gep13 Does that edit on my original post work?

That is good to know more about how it winget works, throwing the idea out there as I don't know any programming just a power user with a feature idea. Wish I could help more but choco rocks keep up the awesome work!

**Comment by pauby** [comment]
I've moved this over to a discussion.
